lib/modules/manager/pip-compile/common.ts

```diff
@@ -27,12 +59,154 @@ export function getPipToolsConstraint(config: UpdateArtifactsConfig): string {
+  .option('--output-file <path>')
+  .option('--extra <extra...>')
+  .option('--extra-index-url <url...>')
+  .allowUnknownOption();
+
+// TODO(not7cd): test on all correct headers, even with CUSTOM_COMPILE_COMMAND
+export function extractHeaderCommand(
+  content: string,
+  fileName: string,
+): PipCompileArgs {
+  const strict: boolean = true; // TODO(not7cd): add to function params
+  const compileCommand = constraintLineRegex.exec(content);
+  if (compileCommand?.groups === undefined) {
+    throw new Error(`Failed to extract command from header in ${fileName}`);
+  }
+  logger.debug(
+    `pip-compile: found header in ${fileName}: \n${compileCommand[0]}`,
+  );
+  const command = compileCommand.groups.command;
+  const argv: string[] = [command];
+  const isCustomCommand = command !== 'pip-compile';
+  if (isCustomCommand) {
+    logger.debug(
+      `pip-compile: custom command ${command} detected (${fileName})`,
+    );
+  }
+  if (compileCommand.groups.arguments) {
+    argv.push(...split(compileCommand.groups.arguments));
+  }
+  logger.debug(
+    `pip-compile: extracted command from header: ${JSON.stringify(argv)}`,
+  );
+  for (const arg of argv) {
+    // TODO(not7cd): check for "--option -- argument" case
+    if (!arg.startsWith('-')) {
+      continue;
+    }
+    throwForDisallowedOption(arg);
+    throwForNoEqualSignInOptionWithArgument(arg);
+    if (strict) {
+      throwForUnknownOption(arg);
+    }
+  }
+
+  // Commander.parse is expecting argv[0] to be process.execPath, pass empty string as first value
+  const parsedCommand = dummyPipCompile.parse(['', ...argv]);
+  const options = parsedCommand.opts();
+  // workaround, not sure how Commander returns named arguments
+  const sourceFiles = parsedCommand.args.filter((arg) => !arg.startsWith('-'));
+  logger.trace(
+    {
+      argv,
+      options,
+      sourceFiles,
+      isCustomCommand,
+    },
+    'Parsed pip-compile command from header',
+  );
+  if (sourceFiles.length === 0) {
+    throw new Error(
+      'No source files detected in command, pass at least one package file explicitly',
+    );
+  }
+  let outputFile = '';
+  if (options.outputFile) {
+    // TODO(not7cd): This file path can be relative like `reqs/main.txt`
+    const file = upath.parse(fileName).base;
+    if (options.outputFile === file) {
+      outputFile = options.outputFile;
+    } else {
+      // we don't trust the user-supplied output-file argument;
+      // TODO(not7cd): allow relative paths
+      logger.warn(
+        { outputFile: options.outputFile, actualPath: file },
+        'pip-compile was previously executed with an unexpected `--output-file` filename',
+      );
+      // TODO(not7cd): this shouldn't be changed in extract function
+      outputFile = file;
+      argv.forEach((item, i) => {
+        if (item.startsWith('--output-file=')) {
+          argv[i] = `--output-file=${quote(file)}`;
```

Comment:
is the `=` mandatory? could it be missing on parsing?

Reply:
Not mandatory for `pip-compile`, but `=` is always in output header. We can widen requirements later, but while we expect valid headers, then let's leave it.